### DISCUSSION TITLE
Help needed in generating join statement for polymorphic many-to-many relationship

### DISCUSSION THREAD

**Discussion by akulmehta** [Power-Components/livewire-powergrid#504] [Q&A]
Hello. Please could someone help with the join statement for the Polymorphic Many-to-Many relationship with the tags example https://laravel.com/docs/9.x/eloquent-relationships#many-to-many-polymorphic-relations

```
posts
    id - integer
    name - string
 
videos
    id - integer
    name - string
 
tags
    id - integer
    name - string
 
taggables
    tag_id - integer
    taggable_id - integer
    taggable_type - string
```

**Comment by luanfreitasdev** [comment]
hello @akulmehta . How did you try to do it? did you have an error?

**Reply by akulmehta** [reply]
Sorry @luanfreitasdev I didn't know even where to begin. An example would be helpful for such a complicated join.

**Reply by akulmehta** [reply]
@luanfreitasdev any help on this - please?

**Reply by luanfreitasdev** [reply]
@akulmehta , for now I haven't worked with polymorphic tables with powergrid.

Can you show how you are doing?

**Reply by akulmehta** [reply]
See above schema, it is taken from laravel docs https://laravel.com/docs/9.x/eloquent-relationships#many-to-many-polymorphic-relations. You have `Post` `Video` and `Tag` models. Since `Tag` can have many-to-many relationship between both `Post` and `Video` therefore, `taggables` table acts as pivot table connecting the three models, where `taggable_type` can be either `Post` or `Video` and `taggable_id` would be corresponding Post or Video id, and the `tag_id` would connect to the corresponding `Tag`. 

Now if we have a table of Posts and we want to show the tags for the posts and allow filtering by tags, we would need to join the data as per PowerGrid requirements. This is the issue I have. I do not know what kind of join statement to write.
